### PR TITLE
feat: parameterize Styles

### PR DESCRIPTION
**Pull request by ludicast**
The `Styles` interface has the `Style` interface as its values.  Sometimes it's useful to generate these types dynamically, so you can take advantage of type hinting.

This PR adds the option for the Styles interface to also allow as its values functions that return a `Style` object.

### CODE REVIEW

File: packages/renderer/index.d.ts [hunk 1]
```diff
@@ -487,7 +487,7 @@ declare namespace ReactPDF {
    * @see https://react-pdf.org/advanced#on-the-fly-rendering
    * @platform web
    */
-  class PDFDownloadLink extends React.Component<PDFDownloadLinkProps> {}
+  class PDFDownloadLink extends React.Component<PDFDownloadLinkProps> { }
```
Comment (diegomura) [inline packages/renderer/index.d.ts]: Can we revert these changes? Was good as it was before too and this is unrelated to PR

**Review by diegomura — COMMENTED** [review]
Looks good! Can we add some tests to make sure this does not break in the future?

File: packages/renderer/index.d.ts [hunk 2]
```diff
@@ -212,8 +212,7 @@ declare namespace ReactPDF {
      * @see https://react-pdf.org/advanced#debugging
      */
     debug?: boolean;
-    href?: string;
-    src?: string;
+    src: string;
```
Comment (diegomura) [inline packages/renderer/index.d.ts:215]: Regression here

File: packages/renderer/index.d.ts [hunk 3]
```diff
@@ -362,9 +361,7 @@ declare namespace ReactPDF {
    */
   class Tspan extends React.Component<React.PropsWithChildren<TspanProps>> {}
 
-  interface GProps extends SVGPresentationAttributes {
-    style?: Style;
-  }
+  interface GProps extends SVGPresentationAttributes {}
```
Comment (diegomura) [inline packages/renderer/index.d.ts:364]: Regression here

File: packages/renderer/index.d.ts [hunk 4]
```diff
@@ -419,11 +416,11 @@ declare namespace ReactPDF {
...
-    cx?: string | number;
-    cy?: string | number;
-    fr?: string | number;
-    fx?: string | number;
-    fy?: string | number;
+    cx: string | number;
+    cy: string | number;
+    fr: string | number;
+    fx: string | number;
+    fy: string | number;
```
Comment (diegomura) [inline packages/renderer/index.d.ts:423]: Ditto

Comment (ludicast) [reply packages/renderer/index.d.ts:423]: yeah, cleaning up & adding tests, going to force push over this
